ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/BuildPropertiesContentAssistProcessor.java

```diff
@@ -44,13 +48,73 @@ public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int
 					continue;
 				}
 				if (element.regionMatches(true, 0, value, 0, value.length())) {
-					TypeCompletionProposal proposal = new TypeCompletionProposal(element, null, element, lineStart,
+					Image img = getImage(element);
+					TypeCompletionProposal proposal = new TypeCompletionProposal(element, img, element, lineStart,
 							value.length());
 					completions.add(proposal);
-				}
 			}
-			return completions.toArray(ICompletionProposal[]::new);
-		} catch (BadLocationException e) {
+		}
+		return completions.toArray(ICompletionProposal[]::new);
+	} catch (BadLocationException e) {
+	}
+	return null;
+}
+
+public Image getImage(String element) {
+	ImageDescriptor desc = null;
+	switch (element)
```

Comment:
The Plugin o.e.pde.ui now requires Java-17, therefore you could convert this to a switch expression and also merge the labels that currently fall-through.

Reply:
Thanks @HannesWell . This has been done. Please have a look.